server/monitor/src/main/java/org/apache/accumulo/monitor/rest/trace/TracesResource.java

```diff
@@ -0,0 +1,372 @@
+    final String principal;
+    final AuthenticationToken at;
+    Map<String,String> loginMap = conf.getAllPropertiesWithPrefix(Property.TRACE_TOKEN_PROPERTY_PREFIX);
+    // May be null
+    String keytab = loginMap.get(Property.TRACE_TOKEN_PROPERTY_PREFIX.getKey() + "keytab");
+    if (keytab == null || keytab.length() == 0) {
+      keytab = conf.getPath(Property.GENERAL_KERBEROS_KEYTAB);
+    }
+
+    if (saslEnabled && null != keytab) {
+      principal = SecurityUtil.getServerPrincipal(conf.get(Property.TRACE_USER));
+      try {
+        traceUgi = UserGroupInformation.loginUserFromKeytabAndReturnUGI(principal, keytab);
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to login as trace user", e);
+      }
+    } else {
+      principal = conf.get(Property.TRACE_USER);
+    }
+
+    if (!saslEnabled) {
+      if (loginMap.isEmpty()) {
+        Property p = Property.TRACE_PASSWORD;
+        at = new PasswordToken(conf.get(p).getBytes(UTF_8));
+      } else {
+        Properties props = new Properties();
+        int prefixLength = Property.TRACE_TOKEN_PROPERTY_PREFIX.getKey().length();
+        for (Entry<String,String> entry : loginMap.entrySet()) {
+          props.put(entry.getKey().substring(prefixLength), entry.getValue());
+        }
+
+        AuthenticationToken token = Property.createInstanceFromPropertyName(conf, Property.TRACE_TOKEN_TYPE, AuthenticationToken.class, new PasswordToken());
+        token.init(props);
+        at = token;
+      }
+    } else {
+      at = null;
+    }
+
+    final String table = conf.get(Property.TRACE_TABLE);
+    Scanner scanner;
+    if (null != traceUgi) {
+      try {
+        scanner = traceUgi.doAs(new PrivilegedExceptionAction<Scanner>() {
+
+          @Override
+          public Scanner run() throws Exception {
+            // Make the KerberosToken inside the doAs
+            AuthenticationToken token = at;
+            if (null == token) {
+              token = new KerberosToken();
+            }
+            return getScanner(table, principal, token);
+          }
+
+        });
+      } catch (IOException | InterruptedException e) {
+        throw new RuntimeException("Failed to obtain scanner", e);
+      }
+    } else {
+      if (null == at) {
+        throw new AssertionError("AuthenticationToken should not be null");
+      }
+      scanner = getScanner(table, principal, at);
+    }
+
+    return new Pair<>(scanner, traceUgi);
+  }
+
+  private Scanner getScanner(String table, String principal, AuthenticationToken at) throws AccumuloException, AccumuloSecurityException {
+    try {
+      Connector conn = HdfsZooInstance.getInstance().getConnector(principal, at);
+      if (!conn.tableOperations().exists(table)) {
+        return null;
+      }
+      return conn.createScanner(table, conn.securityOperations().getUserAuthorizations(principal));
+    } catch (AccumuloSecurityException | TableNotFoundException ex) {
+      return null;
```

Comment:
This is how the previous monitor worked. This class exists for the sole purpose of supporting the AJAX calls. For the trace table, these particular exceptions are normal, because the trace table is not always used, or the trace user is not always configured with the right credentials on every server, or the user may use a different table name for storing traces.

So, it's not clear that monitor visitors should be bombarded with backend error messages in this case. It's sufficient to see that the trace data on the monitor page was not populated.

Also, we don't really have a good/consistent way to propagate internal error messages back up to the monitor's REST interface. Ideally, Jersey would take care of that for us, if we just propagate the exception back up the call stack. But, in the case of the trace table, there's no good reason to.